sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileScanRDD.scala

```diff
@@ -179,7 +182,23 @@ class FileScanRDD(
             currentIterator = readCurrentFile()
           }
 
-          hasNext
+          try {
+            hasNext
+          } catch {
+            case e: SchemaColumnConvertNotSupportedException =>
+              val message = "Parquet column cannot be converted in " +
+                s"file ${currentFile.filePath}. Column: ${e.getColumn}, " +
+                s"Expected: ${e.getLogicalType}, Found: ${e.getPhysicalType}"
+              throw new QueryExecutionException(message, e)
```

Comment:
other changes LGTM too but one question is, why is it `QueryExecutionException`? I thought it'd be `SparkException`.

Reply:
According to my discussion with @gatorsmile, he thought that we should throw some exception that could be captured and displayed in a better form. And since we already have some exception wrapping at https://github.com/apache/spark/blob/master/python/pyspark/sql/utils.py#L60 it seem appropriate to use QueryExecutionException instead of the original SparkException.

Reply:
I don't think that's captured since it's thrown in execution. Can you double check if that's true?

Reply:
Yes, you are right. Sorry, I shouldn't say "use QueryExecutionException instead of the original SparkException". The final exception would still be wrapped with a SparkException. Inside the SparkException would be QueryExecutionException. But the reason is still the same, they don't want to throw too many different Exceptions which might be hard to capture and display.

![38043674-a2485370-326c-11e8-82a6-c36691f1e523](https://user-images.githubusercontent.com/37087310/38722077-9e115878-3eb1-11e8-989c-525268c96e3f.png)